techniques/ios/MASTG-TECH-0066.md

```diff
@@ -6,3 +6,109 @@ platform: ios
 The preferred method of statically analyzing iOS apps involves using the original Xcode project files. Ideally, you will be able to compile and debug the app to quickly identify any potential issues with the source code.
 
 Black box analysis of iOS apps without access to the original source code requires reverse engineering. For example, no decompilers are available for iOS apps (although most commercial and open-source disassemblers can provide a pseudo-source code view of the binary), so a deep inspection requires you to read assembly code.
+
+## Using @MASTG-TOOL-0073
+
+Radare2 (r2) is a powerful reverse engineering framework that provides a wide range of tools for analyzing binaries. This section explains how to use r2 effectively for static analysis, focusing on identifying and analyzing relevant functions in iOS applications.
+
+For interactive analysis, you can also use radare2's [visual mode](https://book.rada.re/visual/visual_disassembly.html) by pressing `V` after opening a binary, which provides a more graphical interface for navigation and analysis.
+
+### Identifying Relevant Functions
+
+**Listing Functions**:
+
+Use the `afl` command to [list all functions](https://book.rada.re/analysis/code_analysis.html#analyze-functions) in the binary and combine the [`~` operator](https://book.rada.re/first_steps/syntax.html#filtering) to filter functions by name. For example, if you're interested in cryptographic functions, you can search using a filter like `afl~SecKey` to find all functions related to `SecKey`. If you already know the function name, you can use `afl~<function_name>` to directly filter for that specific function.
+
+```bash
+afl~SecKeyCreateRandomKey
+```
+
+Example: In @MASTG-DEMO-0011, `SecKeyCreateRandomKey` is identified as a function of interest.
+
+**When Not to Use `afl`**:
+
+In some cases, functions may not appear in the list generated by `afl`. For example, `canEvaluatePolicy` is a dynamically resolved function and may not be listed.
+
+```bash
+afl~canEvaluatePolicy
+< no results >
+```
+
+Instead, use [flags](https://book.rada.re/commandline/flags.html) (`f`), string searches with `/`, `/z`, `iz~` and then use [cross-references](#analyzing-cross-references) (`axt`) to locate its usage in the binary. This approach ensures that dynamically resolved or obfuscated functions are not overlooked.
+
+```bash
+f~canEvaluatePolicy
+0x100008360 24 str.canEvaluatePolicy:error:
+```
+
+In the output, you can see a flag `str.canEvaluatePolicy:error:` representing the string `canEvaluatePolicy:error:` which is located at `0x100008360`. You can then use the [`axt` command](https://book.rada.re/analysis/code_analysis.html#analyze-xrefs) to find cross-references to this string using its address:
+
+```bash
+axt @ 0x100008360
```

Comment:
this command also takes the destination address as argument, so, "axt 0x10000360" is the equivalent without the temporal seek.

it's worth to mention that @ and `s` also take partial addresses, so you dont need to type the whole hexadecimal number. just typing "axt@..8360` would be enough too

Reply:
also ax~8360 can be helpful, as well as using the "ax," command